### ISSUE TITLE
Koreader android - way to toggle immersive mode / disable status bar toggle for Android 6+?

### ISSUE THREAD

**Issue by Eevoo** [koreader/koreader#9100]
Hello all. Thanks for making such an excellent reader app for phones and ereaders!

TL;DR: can there be a settings toggle to disable the app default immersive mode? The default value is great for e-readers like my Likebook Mars, not as great for my Samsung.

I'm requesting basically #2902 but just for newer forms of Android (6+ so API 23 through 33?) instead of [for APIs 14-18](https://github.com/koreader/koreader/blob/master/frontend/ui/elements/common_settings_menu_table.lua#L307) (which AFAIU stops at Android 4.3), is there a way to make the fullscreen toggle work for android versions 6+ and higher?

I think I found where [fullscreen is toggled](https://github.com/koreader/koreader/blob/ea3fa5c2c7ed7330e8e1e6ef838b780e7998ea51/frontend/ui/elements/screen_android.lua#L65) but found that the [underlying function](https://github.com/koreader/koreader/blob/217a73f3c0be5832d2ccd289eb0f1d8528e3cd75/frontend/device/android/device.lua#L345) has been stubbed out, apparently because koreader is using immersive mode by default based on [this potentially unoffical forum post](https://www.mobileread.com/forums/showpost.php?p=3820028&postcount=222)?

Before I discovered said forum post, I googled around for ways to easily update the fullscreen toggle and found this [Stack Overflow Answer](https://stackoverflow.com/a/31439262/11351150) came up. Checking the [Android developer documentation that the answer referenced, the WindowManager.LayoutParams.FLAG_FULLSCREEN flag](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#FLAG_FULLSCREEN) has been deprecated in API level 30 with alternatives listed in the answer, but I stopped following that rabbit trail when I discovered immersive mode has been enabled by default for the app.

Sorry if I'm looking in the wrong areas for these settings, I'm not fully able to map whether the Device.firmware_rev shown [here](https://github.com/koreader/koreader/blob/master/frontend/device/android/device.lua#L17) closely maps the targetSdkVersion or the compiledSdkVersion or neither or both - I'm neither a lua programmer nor an Android programmer but I tried to google around a bit so I could give more info up front.

**Comment by pazos** [comment]
Duplicate of https://github.com/koreader/koreader/issues/8532

@Eevoo please see the info in the linked ticket.

BTW, mobileread is the official forum for us :)

**Comment by Eevoo** [comment]
Thanks for the pointer! I googled for this but somehow didn't locate that ticket.
